### PR TITLE
Feature/ubi 342 friendslist v1

### PR DESCRIPTION
**Pull request by MaximeStOnge**
#### What changes does this PR contain?
First iteration of friendslist

#### Which JIRA ticket ID's are associated with this PR?
UBI-342

#### Does this PR change the outward facing API?
yes

#### Additional comments and information

@ubivius/microservices

### CODE REVIEW

File: pkg/data/relationships.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+// ErrorUserNotFound : User specific errors
+var ErrorUserNotFound = fmt.Errorf("UserID doesn't exist")
+
+// RelationshipType of a relationship
+type RelationshipType int
+
+// relationship type of a friend request
+const (
+    None RelationshipType = iota  // user has no intrinsic relationship
```
Comment (ArcticWalrus) [inline pkg/data/relationships.go:25]: Alignment is weird here

File: pkg/data/relationships.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    Friend						  // user is a friend
+    Blocked						  // user is blocked
+	PendingIncoming				  // user has a pending incoming friend request to connected user
+	PendingOutgoing				  // current user has a pending outgoing friend request to user
+)
+
+// Relationship defines the structure for an API relationship.
+// Formatting done with json tags to the right. "-" : don't include when encoding to json
```
Comment (ArcticWalrus) [inline pkg/data/relationships.go:33]: remove comment on json tags
Also alignment is weird again

File: pkg/data/relationships.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+}
+
+// Returns an array of friends invites in the database
+// Returns -1 when no relationship is found
+func findInvitesListByUserID(id int) Relationships {
+	var invitesList Relationships
+	for _ , relationship := range relationshipList {
+		if relationship.User1.UserID == id && relationship.User1.RelationshipType == PendingIncoming {
+			invitesList = append(invitesList, relationship)
+		}else if relationship.User2.UserID == id && relationship.User2.RelationshipType == PendingIncoming{
```
Comment (ArcticWalrus) [inline pkg/data/relationships.go:131]: space before else

File: pkg/data/validate.go [hunk 4]
```diff
@@ -1,35 +1,29 @@
...
-// The struct to validate should be passed as an interface in the future and the errors should be handled as individual error strings
-// For further information see :
-// Validator library : https://github.com/go-playground/validator
-// Nic Jackson episode : https://github.com/nicholasjackson/building-microservices-youtube/blob/episode_7/product-api/data/validation.go
+// ErrorInvalidRelationshipType : Invalid RelationshipType specific error
+var ErrorInvalidRelationshipType = fmt.Errorf("Invalid RelationshipType")
 
-// ValidateProduct a product with json validation and customer SKU validator
-func (product *Product) ValidateProduct() error {
+// ValidateRelationship a relationship with json validation and customer SKU validator
```
Comment (ArcticWalrus) [inline pkg/data/validate.go:12]: Customer sku?

**Review by ArcticWalrus — CHANGES_REQUESTED** [review]

**Comment by ArcticWalrus** [comment]
relationshiptype should probably be a string. When I do the curls I have no idea what the data means

File: pkg/data/relationships.go (hunk 2 above)
Comment (ArcticWalrus) [reply pkg/data/relationships.go:33]: Make the json value user_1 and user_2
conversation_id

File: pkg/data/relationships.go [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+	User2 			User	`json:"user2"`
+	ConversationID 	int 	`json:"conversationid"`
+	CreatedOn   	string  `json:"-"`
+	UpdatedOn   	string  `json:"-"`
+	DeletedOn   	string  `json:"-"`
+}
+
+// User in a relationship
+type User struct {
+	UserID      		int  	     		`json:"userid" validate:"required"`
```
Comment (ArcticWalrus) [inline pkg/data/relationships.go:46]: in json user_id
relationship_type
Unreadable otherwise in json strings

File: README.md [hunk 6]
```diff
@@ -1,9 +1,2 @@
-# microservice-template
```
Comment (ArcticWalrus) [inline README.md:1]: Add description of the endpoints in readme. I'm not sure what each endpoint does just from reading the urls

**Review by ArcticWalrus — CHANGES_REQUESTED** [review]

**Review by ArcticWalrus — APPROVED** [review]

**Review by JeremiSavard — APPROVED** [review]

**Pull request merged**